lib/asset/assetmultipart.js

```diff
@@ -39,13 +40,13 @@ class AssetMultipart {
      */
     constructor(targetUrls, minPartSize, maxPartSize, headers, completeUrl, uploadToken) {
         if (!Array.isArray(targetUrls) || (targetUrls.length === 0)) {
-            throw Error(`Invalid target urls: ${targetUrls}`);
+            throw new IllegalArgumentError("Invalid target urls", targetUrls);
```

Comment:
For this one it would be good to elaborate the error text - because from the text you can't know if the issue is that the URLs are not valid (expired?) or that there were no URLs provided.